Modules/Retailer/Http/Controllers/RetailerController.php

```diff
@@ -63,5 +64,256 @@ public function register(Request $request)
+                        $country = DB::table('countries')->where('id', $order->country)->first();
+                        $shippingDetailsArr[] = $country->name;
+                    }
+                    $shippingDetailsStr = implode(',', $shippingDetailsArr);
+                    switch ($order->status) {
+                        case 'new':
+                            $orderStatus = 'open';
+                            break;
+                        case 'unfulfilled':
+                            $orderStatus = 'in progress';
+                            break;
+                        case 'fulfilled':
+                            $orderStatus = 'completed';
+                            break;
+                        default:
+                            $orderStatus = $order->status;
+                            break;
+                    }
+                    if (!empty($cart)) {
+                        foreach ($cart as $cItem) {
+                            $product = Products::find($cItem->product_id);
+
+                            $rorders[] = array(
+                                'product_name' => $cItem->product_name,
+                                'product_image' => $product->featured_image != '' ? $product->featured_image : asset('public/img/logo-image.png'),
+                                'style_name' => $product->style_name,
+                                'style_group_name' => $product->style_group_name,
+                                'brand' => $brand->brand_name,
+                                'order_number' => $order->order_number,
+                                'order_id' => $order->id,
+                                'total_amount' => $order->total_amount,
+                                'order_date' => date('Y-m-d', strtotime($order->created_at)),
+                                'customer_name' => $order->name,
+                                'order_status' => $orderStatus,
+                                'shipping_date' => $order->shipping_date,
+                                'shipping_details' => $shippingDetailsStr,
+                            );
+                        }
+                    }
+                }
+            }
+            $data = array(
+                "orders" => $rorders,
+                "allOrdersCount" => $allOrdersCount,
+                "newOrdersCount" => $newOrdersCount,
+                "unfulfilledOrdersCount" => $unfulfilledOrdersCount,
+                "fulfilledOrdersCount" => $fulfilledOrdersCount,
+                "cancelledOrdersCount" => $cancelledOrdersCount
+            );
+            $response = ['res' => true, 'msg' => "", 'data' => $data];
+            return response()->json($response);
+        } else {
+            $response = ['res' => false, 'msg' => "No record found", 'data' => ""];
+            return response()->json($response);
+        }
+    }
+
+    public function order(Request $request, $order_number)
```

Comment:
Why do you have a duplicate function (override)? If this function is taking $orderNumber as additional parameter then this can be achieved within the above order function by setting the second function parameter as null i.e. `    public function order(Request $request, $order_number = null)`